autoscaler/controllers/common/processors.go

```diff
@@ -30,3 +36,52 @@ func FilterAndSortProcessorsByOrderHint(processors *odigosv1.ProcessorList, coll
 
 	return filteredProcessors
 }
+
+func DeleteProcessorByType(ctx context.Context, c client.Client, processorType string, namespace string) error {
+	var processors odigosv1.ProcessorList
+	if err := c.List(ctx, &processors, client.InNamespace(namespace)); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	if processor := GetProcessorIfExists(&processors, processorType); processor != nil {
+		return c.Delete(ctx, processor)
+	}
+
+	return nil
+}
+
+func GetProcessorIfExists(allProcessors *odigosv1.ProcessorList, processorType string) *odigosv1.Processor {
+	for _, processor := range allProcessors.Items {
+		if processor.Spec.Type == processorType {
+			return &processor
+		}
+	}
+	return nil
+}
+
+func GetGenericBatchProcessor() odigosv1.Processor {
+	emptyConfig, _ := json.Marshal(make(map[string]interface{}))
+
+	return odigosv1.Processor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "generic-batch-processor",
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "odigos.io/v1alpha1",
+			Kind:       "Processor",
+		},
+		Spec: odigosv1.ProcessorSpec{
+			Type: "batch",
+			CollectorRoles: []odigosv1.CollectorsGroupRole{
+				odigosv1.CollectorsGroupRoleClusterGateway,
+				odigosv1.CollectorsGroupRoleNodeCollector},
+			OrderHint:       0,
```

Comment:
We should apply the probabilistic sampler before the batch, otherwise we store things in memory just to drop them right after